app/presenters/candidate_interface/application_form_presenter.rb

```diff
@@ -73,6 +73,13 @@ def application_choice_errors
             )
             next
           end
+
+          if choice.course_closed_on_apply?
+            error_list << ApplicationChoiceError.new(
+              choice.course_closed_on_apply_error, choice.id
+            )
+            next
```

Comment:
I wonder if it's worth making this the 1st or 2nd check - I'm thinking that if a course is closed on Apply, its course-level or site-level vacancy status may remain unchanged, in which case moving this block further up saves us a couple of unnecessary SQL queries.